ads/google/a4a/utils.js

```diff
@@ -113,6 +114,26 @@ export const TRUNCATION_PARAM = {name: 'trunc', value: '1'};
 const CDN_PROXY_REGEXP =
   /^https:\/\/([a-zA-Z0-9_-]+\.)?cdn\.ampproject\.org((\/.*)|($))+/;
 
+/** @const {string} */
+const TOKEN_VALUE =
+  'A8Ujr8y+9sg/ZBmCs90ZfQGOUFJsAS/YaHYtjLAsNn05OaQXSmZeRZ2U1wAj3PD74WY9re2x/TwinJoOaYuqFQoAAACBeyJvcmlnaW4iOiJodHRwczovL2FtcHByb2plY3QubmV0OjQ0MyIsImZlYXR1cmUiOiJDb252ZXJzaW9uTWVhc3VyZW1lbnQiLCJleHBpcnkiOjE2MzE2NjM5OTksImlzU3ViZG9tYWluIjp0cnVlLCJ1c2FnZSI6InN1YnNldCJ9';
+
+/**
+ * Inserts origin-trial token for `attribution-reporting` if not already
+ * present in the DOM.
+ * @param {!Window} win
+ */
+export function maybeInsertOriginTrialToken(win) {
+  if (win.document.querySelector(`meta[content='${TOKEN_VALUE}']`)) {
+    return;
+  }
+  const metaEl = createElementWithAttributes(win.document, 'meta', {
+    'http-equiv': 'origin-trial',
+    content: TOKEN_VALUE,
+  });
+  win.document.head.appendChild(metaEl);
```

Comment:
Did you verify this actually works? I'd be extremely surprised.

Reply:
Could you clarify which part you don't think will work? Maybe I missed something. Adding OT with script is supported http://googlechrome.github.io/OriginTrials/developer-guide.html#16-can-i-provide-tokens-by-running-script

Reply:
Possibly for `querySelector` a very long `TOKEN_VALUE` might exceed the length limit?

Reply:
Surprising, I didn't realize OTs could be inserted via JS.